core/src/main/java/org/kiwix/kiwixmobile/core/page/bookmark/viewmodel/effects/ShowDeleteBookmarksDialog.kt

```diff
@@ -36,7 +36,11 @@ data class ShowDeleteBookmarksDialog(
   @Inject lateinit var dialogShower: DialogShower
   override fun invokeWith(activity: AppCompatActivity) {
     (activity as BookmarksActivity).activityComponent.inject(this)
-    dialogShower.show(DeleteBookmarks, {
+    var dialogType: KiwixDialog = KiwixDialog.DeleteAllBookmarks
+    if (state.isInSelectionState) {
+      dialogType = KiwixDialog.DeleteSelectedBookmarks
+    }
+    dialogShower.show(dialogType, {
       effects.offer(DeleteBookmarkItems(effects, state, bookmarksDao))
```

Comment:
hmm I must have glossed over this originally, this being an action (UserConfirmedDeletion perhaps) would be better, tiny bit out of scope but we can boyscout this I think

Reply:
It used to be an action but I changed it into an effect because if performDeleteAction is an action, the order of operations is :

```
fun performDeleteAction(state: State){
  dao.deleteItems(state.itemsToDelete)
  return state
}
```

The dao call causes the `daoIsUpdated` action to be called before the return of the pre-deleted state is returned. This means the state will first be updated with the deleted items removed from it, but then being replaced with the pre-deleted state. I made the `deleteITems` into a side-effect because then the state will only update once, because of the dao being updated.

Reply:
Why is the pre-deleted state returned? You can put a distinctUntilChanged() on some emissions if they are triggering excessively.
It should go clickDelete->showDialogEffect->confirmDelete->deleteItemsEffect->DBUpdate changes state, items are removed, nothing is in selection, action mode finishes

Reply:
It's returned by the `deleteRequestedItems()` function that is called from the action.

```
  private fun deleteRequestedItems(state: BookmarkState): BookmarkState {
    effects.offer(DeleteBookmarkItems(effects, state, bookmarksDao))
    return state <<<<<<<<<<<<<<< here
  }
```

Seems to be a race condition as it updates wrongly only sometimes. 

Option 1, the good one: 
clickDelete->showDialogEffect->confirmDelete->deleteItemsEffect->pre-delete state return->DBUpdate-> DBUpdate state return

Option 2, the bad one
clickDelete->showDialogEffect->confirmDelete->deleteItemsEffect->DBUpdate->DBUpdate state return ->pre-delete state return

![untitled](https://user-images.githubusercontent.com/22193232/85993055-0a0dee80-b9f6-11ea-8832-0a4d687f4414.gif)